File: legacy_node/controllers/booking_controller.js
import Booking from "../models/booking.js"
import { isAvailable } from "../services/availability_service.js"
import { calculateCost } from "../services/pricing_service.js"

export const createBooking = async (req, res) => {
  const { space_id, start_date, end_date, duration } = req.body

  const available = await isAvailable(space_id, start_date, end_date)
  if (!available) {
    return res.status(400).json({ error: "space not available" })
  }

  const total_cost = calculateCost(duration, 100)

  const booking = await Booking.create({
    user_id: req.user.id,
    space_id,
    start_date,
    end_date,
    duration,
    total_cost,
  })

  res.status(201).json(booking)
}

export const checkAvailability = async (req, res) => {
  const { spaceId } = req.params
  const { start_date, end_date } = req.query

  const available = await isAvailable(spaceId, start_date, end_date)
  res.json({ available })
}

File: server/app/controller/booking_controller.py
from flask import request, jsonify
from app.extensions import db
from app.models.booking import Booking
from app.services.availability_service import check_availability

def check_availability_controller(space_id):
    start_date = request.args.get("start_date")
    end_date = request.args.get("end_date")

    available = check_availability(space_id, start_date, end_date)
    return jsonify({"available": available})

def create_booking_controller():
    data = request.get_json()

    booking = Booking(
        user_id=1,  # replace with JWT user later
        space_id=data["space_id"],
        start_date=data["start_date"],
        end_date=data["end_date"],
        duration=data["duration"],
        total_cost=100
    )

    db.session.add(booking)
    db.session.commit()

    return jsonify({"message": "Booking created"}), 201
